vllm/model_executor/models/llama.py

```diff
@@ -386,6 +386,52 @@ def sample(
         next_tokens = self.sampler(logits, sampling_metadata)
         return next_tokens
 
+    def _convert_weights_to_fp8(
+            self, state_dict: Dict[str, torch.tensor],
+            weights_to_convert: List[str]) -> Dict[str, torch.tensor]:
+        """Converts the original weights to FP8E4M3 from FP16."""
+        for weight_name in weights_to_convert:
+            weight_scale_name = weight_name + "_scale"
+            if weight_scale_name not in state_dict:
+                continue
+            loaded_weight = state_dict[weight_name]
+            scale = state_dict[weight_scale_name]
+            state_dict[weight_name] = (loaded_weight.cpu() / scale.cpu()).to(
+                torch.float8_e4m3fn)
+        return state_dict
+
+    def _convert_scales_for_vllm(self, key, value):
+        """Replaces the names of *quantizer._amax to _scale."""
+        replacements = {
+            "weight_quantizer._amax": "weight_scale",
+            "input_quantizer._amax": "act_scale",
```

Comment:
This has been updated such that act_scale -> input_scale https://github.com/vllm-project/vllm/pull/5353